pylint/config/option_manager_mixin.py

```diff
@@ -87,13 +87,7 @@ def reset_parsers(self, usage=""):
 
     def register_options_provider(self, provider, own_group=True):
         """Register an options provider."""
-        assert provider.priority <= 0, "provider's priority can't be >= 0"
-        for i, options_provider in enumerate(self.options_providers):
-            if provider.priority > options_provider.priority:
-                self.options_providers.insert(i, provider)
-                break
-        else:
-            self.options_providers.append(provider)
+        self.options_providers.append(provider)
```

Comment:
Option provider priority is different than checker priority, right? We should document that as going away also. 

I'm a little nervous that a behavior change could be hiding here, but maybe you can reassure me :D. I tried to trace upstream from this where this could be used and noticed `ConfigurationMixin` (subclass of `OptionsManagerMixIn`) is used in pyreverse, which is way outside my involvement.

Also, apparently checkers are option providers? So could removing the checker priority alter the status quo for what order checker-provided-optoins get overwritten?  But if this is part of removing complexity during the argparse migration that's fine...

Reply:
Well, since each check is a option provider they share the same priority. Thus, we're basically doing the same checks here as we're doing at that other place.

I think the only two "functions" that come from these priorities is that:
1) Checkers will be run sooner, but when is that ever needed?
2) Options _could_ be overwritten by providers with higher priority. But then again, why would we allow that? That creates a lot of mess and duplicate options have actually caused me multiple headaches before.